scarb/src/core/registry/client/cache.rs

```diff
@@ -145,7 +145,7 @@ impl<'c> RegistryClientCache<'c> {
             let tarball_name = package.tarball_name();
             let file = self
                 .dl_fs
-                .open_rw(&tarball_name, &tarball_name, self.config)?;
+                .open_ro(&tarball_name, &tarball_name, self.config)?;
```

Comment:
This is necessary as `open_rw` truncates the existing contents of the file, effectively destroying the cached tarball: 

>    /// This function will create a file at `path` if it doesn't already exist (including
>    /// intermediate directories) else if it does exist, it will be truncated.
>    ...
>    pub fn open_rw(

Since all tests pass and `scarb fetch`, `scarb build` work without issue, write access might not be necessary when accessing a cached archive. We could consider creating an alternative method that provides write access without truncating the file, but unless there's a specific need for this, it seems to me that read-only access should be sufficient in this context.

Reply:
Also, it's not very intuitive from the naming that `open_rw` is supposed to truncate anything 🤔

Reply:
You can fire git blame and see that truncating was added after this function was created :) There was a relatively significant amount of consideration going into that name and the conclusion was that in Scarb context: you will (almost) never want to write to a file without overwriting it.

Reply:
> You can fire git blame and see that truncating was added after this function was created :) 

Thanks for clarification! Although I assume the discussion happened offline rather then in the [PR](https://github.com/software-mansion/scarb/pull/388).

> you will (almost) never want to write to a file without overwriting it.

I understand the rationale behind *typically* wanting to overwrite files, but I think "almost" is the key and the difference between intended and actual use can lead to bugs that can tricky to spot 😅 . Maybe we should consider changing it to something like `create_open_rw` or `create_rw`, just to fully convey what this does and be safe in the future. Thoughts?

Reply:
I'm fine with that

Reply:
This can be a separate PR, please merge this one 🙏